### ISSUE TITLE
Amazon S3 - Missing error handling in TransferService results in NPE when resuming transfer

### ISSUE THREAD

**Issue by hfcheng** [aws/aws-sdk-android#299]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?
Amazon S3
* In what version of SDK are you facing the problem?
2.4.0
* Is the issue limited to Simulators / Actual Devices?
Happens on actual devices, not tested on simulator
* Can your problem be resolved if you bump to a higher version of SDK?
No
* Is this problem related to specific Android/OS version?
No
* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.
There is an error handling miss inside `execCommand(Intent intent)` in TransferService.java when handling Intent with type **INTENT_ACTION_TRANSFER_RESUME**. In some exceptional cases when the SDK cannot find correspond Transfer from DB, current implementation still proceed with the start() call and results in NPE.  Proper error handling are implemented for CANCEL and PAUSE but not RESUME.

* Please include a stacktrace if applicable.
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.amazonaws.mobileconnectors.s3.transferutility.TransferRecord.start(com.amazonaws.services.s3.AmazonS3, com.amazonaws.mobileconnectors.s3.transferutility.TransferDBUtil, com.amazonaws.mobileconnectors.s3.transferutility.TransferStatusUpdater, com.amazonaws.mobileconnectors.s3.transferutility.TransferService$NetworkInfoReceiver)' on a null object reference
       at com.amazonaws.mobileconnectors.s3.transferutility.TransferService.execCommand(Unknown Source)
       at com.amazonaws.mobileconnectors.s3.transferutility.TransferService$UpdateHandler.handleMessage(Unknown Source)
       at android.os.Handler.dispatchMessage(Handler.java:111)
       at android.os.Looper.loop(Looper.java:194)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```
If you need help with understanding how to implement something in particular then we suggest that you first look into our [developer guide](https://docs.aws.amazon.com/mobile/sdkforandroid/developerguide/). You can also simplify your process of creating an application by using [Mobile Hub](https://console.aws.amazon.com/mobilehub/home#/).

**Comment by mlshon** [comment]
Thank you for bringing this to our attention. We will update this thread when we make the fix.

**Comment by karthiksaligrama** [comment]
resolved with 2.4.3
